### PR TITLE
[AIRFLOW-XXXX] Add scheduler in production section to best practices doc

### PR DESCRIPTION
**Pull request by ryw**
---
Issue link: `Document only change, no JIRA issue`

Make sure to mark the boxes below before creating PR: [x]

- [X] Description above provides context of the change
- [X] Commit message/PR title starts with `[AIRFLOW-NNNN]`. AIRFLOW-NNNN = JIRA ID<sup>*</sup>
- [X] Unit tests coverage for changes (not needed for documentation changes)
- [X] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [X] Relevant documentation is updated including usage instructions.
- [X] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

<sup>*</sup> For document-only changes commit message can start with `[AIRFLOW-XXXX]`.

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

File: docs/best-practices.rst [hunk 1]
```diff
@@ -315,3 +315,16 @@ Some configurations such as Airflow Backend connection URI can be derived from b
...
+Scheduler Uptime
+-----------------
+
+Airflow users have for a long time been affected by a
+`core Airflow bug <https://issues.apache.org/jira/browse/AIRFLOW-401>`_
+that causes the scheduler to hang without a trace.
+
+Until fully resolved, you can mitigate a few ways:
+
+* Set a reasonable run_duration setting in your `airflow.cfg`. `See example <https://github.com/astronomer/airflow-chart/blob/63bc503c67e2cd599df0b6f831d470d09bad7ee7/templates/configmap.yaml#L44>`_.
```
Comment (kaxil) [inline docs/best-practices.rst]:
```suggestion
* Set a reasonable run_duration setting in your ``airflow.cfg``. `See example <https://github.com/astronomer/airflow-chart/blob/63bc503c67e2cd599df0b6f831d470d09bad7ee7/templates/configmap.yaml#L44>`_.
```

File: docs/best-practices.rst [hunk 2]
```diff
@@ -315,3 +315,16 @@ Some configurations such as Airflow Backend connection URI can be derived from b
...
+* Add an `exec` style health check to your helm charts on the scheduler deployment to fail if the scheduler has not heartbeat in a while. `See example <https://github.com/astronomer/helm.astronomer.io/pull/200/files>`_.
```
Comment (kaxil) [inline docs/best-practices.rst]:
```suggestion
* Add an ``exec`` style health check to your helm charts on the scheduler deployment to fail if the scheduler has not heartbeat in a while. `See example <https://github.com/astronomer/helm.astronomer.io/pull/200/files>`_.
```

**Comment by ryw** [comment]
@kaxil thanks for suggestions - i accepted your changes, then merged back into one commit + forced pushed to my branch. is that the right way?

**Comment by ryw** [comment]
@kaxil hmm looks like random test fail?

**Comment by potiuk** [comment]
Thanks @ryw !

**Comment by kaxil** [comment]
> @kaxil thanks for suggestions - i accepted your changes, then merged back into one commit + forced pushed to my branch. is that the right way?

Yup :)

**Pull request merged**